### ISSUE TITLE
DSL preview doesn't load classes in a composite build

### ISSUE THREAD

**Issue by jeancharles-roger** [LouisCAD/Splitties#130]
Hi, 

When using a class imported from a composite build in gradle, the DSL preview can't load the class.

```
java.lang.ClassNotFoundException: com.mycompany.task.Test
	at org.jetbrains.android.uipreview.ModuleClassLoader.load(ModuleClassLoader.java:189)
	at com.android.tools.idea.rendering.RenderClassLoader.findClass(RenderClassLoader.java:61)
	at org.jetbrains.android.uipreview.ModuleClassLoader.findClass(ModuleClassLoader.java:123)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at org.jetbrains.android.uipreview.ModuleClassLoader.loadClass(ModuleClassLoader.java:222)
	at com.mycompany.app.TestUi.<init>(TestUi.kt:47)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at splitties.viewdsl.idepreview.UiPreView.<init>(UiPreView.kt:73)
	at splitties.viewdsl.idepreview.UiPreView.<init>(UiPreView.kt:52)
	at splitties.viewdsl.idepreview.UiPreView.<init>(UiPreView.kt)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.jetbrains.android.uipreview.ViewLoader.createNewInstance(ViewLoader.java:401)
	at org.jetbrains.android.uipreview.ViewLoader.loadClass(ViewLoader.java:184)
	at org.jetbrains.android.uipreview.ViewLoader.loadView(ViewLoader.java:142)
	at com.android.tools.idea.rendering.LayoutlibCallbackImpl.loadView(LayoutlibCallbackImpl.java:229)
	at android.view.BridgeInflater.loadCustomView(BridgeInflater.java:421)
	at android.view.BridgeInflater.loadCustomView(BridgeInflater.java:432)
	at android.view.BridgeInflater.createViewFromTag(BridgeInflater.java:336)
	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:730)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:492)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:394)
	at com.android.layoutlib.bridge.impl.RenderSessionImpl.inflate(RenderSessionImpl.java:327)
	at com.android.layoutlib.bridge.Bridge.createSession(Bridge.java:386)
	at com.android.tools.idea.layoutlib.LayoutLibrary.createSession(LayoutLibrary.java:193)
	at com.android.tools.idea.rendering.RenderTask.createRenderSession(RenderTask.java:450)
	at com.android.tools.idea.rendering.RenderTask.lambda$inflate$3(RenderTask.java:590)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

com.mycompany.task.Test is loaded from an included build. The preview works well when no classes from the composite build is referenced.

I'm not sure this issue can be addressed here but I've search on Android Studio but I can't find any reference to this problem. 

Best regards

**Comment by jeancharles-roger** [comment]
I've tested the integration with the bundled jar, it works like a charm. It's clearly related to composite build

**Comment by LouisCAD** [comment]
Hi Jean-Charles!
I'm glad to see I'm not the only one using this work! (I hadn't got any usage feedback on Splitties View DSL until now)

Please, could you make a custom View that uses the `com.mycompany.task.Test`, then write an xml layout file that uses this custom view, and tell me if you're experiencing the same error?

And yes, the issue is likely related to composite builds. It's likely to be an Android Studio or Android Gradle Plugin related issue which you can report on [b.android.com](http://b.android.com) (and link back here).

**Comment by jeancharles-roger** [comment]
Hi, 

Your viewdsl helps me quite a lot to swiftly create nice api driven layouts. I've tested anko before but I was missing the ability to use variables/fields to add views.

I've created an XML layout with a custom view referencing a class in a composite build, the error is the same, splitties isn't the issue here.

I've filled a bug to Android Studio team: https://issuetracker.google.com/issues/117581682

Thanks for your help
